backend/src/plugins/fastifyRowLockCheck/index.js

```diff
@@ -0,0 +1,56 @@
+const dbLockModel = require("@models/db_lock");
+const fp = require("fastify-plugin");
+
+const fastifyRowLockCheck = async (fastify, options) => {
+  let dbRowLock = {};
+
+  const getRowLock = async (request, reply) => {
+    const requestData = {
+      locked_row_id: Number(request.headers.locked_row_id),
+      locked_table: request.headers.locked_table,
+      locked_by: request.headers.locked_by,
+    };
+
+    dbLockModel
+      .getLockByParams(requestData)
+      .then((lockInfo) => {
+        if (lockInfo) {
+          dbRowLock = {
+            locked: true,
+            locked_by: lockInfo.locked_by,
+            currentUser: lockInfo.locked_by === requestData.locked_by ? true : false,
+            lockId: lockInfo.id,
```

Comment:
We're mixing camel case and snake case here. I can't remember which case we use for responses but we should stick to that.

Reply:
Yes this is because the database is snake case and the backend is in JS